src/manifold/src/manifold.cpp

```diff
@@ -138,6 +138,52 @@ Mesh Manifold::GetMesh() const {
   return result;
 }
 
+size_t Manifold::GetMeshDirect(int mode) const {
+  const Impl& impl = *GetCsgLeafNode().GetImpl();
+
+  int recs = 8;
+  int nv = NumVert();
+  int nt = NumTri();
+
+  if (mode == 1) {
+    recs += (nt * 3);
+  } else if (mode == 2) {
+    recs += (nt * 9);
+  }
+
+  int ms = sizeof(size_t) * recs;
+  size_t* x = (size_t*)malloc(ms);
+  float* f = (float*)x;
+
+  int rp = 0;
+  // fixed header
+  x[rp++] = (size_t)impl.vertPos_.begin();
+  x[rp++] = (size_t)impl.vertPos_.end();
+  x[rp++] = nv;
+  x[rp++] = (size_t)impl.halfedge_.begin();
+  x[rp++] = (size_t)impl.halfedge_.end();
+  x[rp++] = nt;
+  x[rp++] = (size_t)impl.vertNormal_.begin();
+  x[rp++] = (size_t)impl.vertNormal_.end();
```

Comment:
How about avoiding malloc and free using some modern constructs?
```
struct MeshGL {
  std::array<float> vertPos;
  std::array<uint> triVerts;
  std::array<float> vertNormal;
}
```

Reply:
Yes, the WASM side will need to call `delete()`, which will then invoke the default destructor of `std::array` to free the memory. I believe there will always be a copy somewhere - part of why we don't expose our internal data structures is because on some platforms they reside on the GPU. We just need to make that copy efficient. At the end of the day these buffer copies (at least in C++) are dramatically faster than the operations this library is doing anyway. For this reason I don't want to worry about things like conditionally including normals - I think it's more API complexity than it's worth. 

I think copying our buffers to the right form in the C++ function and then using a `subarray` to pull a `std::array.data()` pointer into a `Float32Array` in JS should give us most of the speedup we need. If I'm wrong though, please show me the bench marks!

Reply:
Understood wrt GPU and internal data. I hope this can be approached iteratively as this may require C++ skills not currently in my inventory and the gains are important to CAM simulation work.

Reply:
I would very much like to be able to construct a Manifold.Mesh directly from arrays, as well. But it was not clear to me how to do that.

Reply:
> I would very much like to be able to construct a Manifold.Mesh directly from arrays, as well. But it was not clear to me how to do that.

that could be very useful :) ... I am not sure exactly, but sound like a nice option to have

Reply:
Agreed, once we settle on some structure like `MeshGL`, we should enable it for both input and output.